lfsapi/schema/reader.go

```diff
@@ -0,0 +1,108 @@
+package schema
+
+import (
+	"bytes"
+	"io"
+	"strings"
+	"sync/atomic"
+
+	"github.com/git-lfs/git-lfs/errors"
+	"github.com/xeipuuv/gojsonschema"
+)
+
+var (
+	// ErrNoValidationDone is an error returned when `ValidationErr()` is
+	// called before having processed any data.
+	ErrNoValidationDone = errors.New("lfsapi/schema: no validation done")
+	// ErrValidationIncomplete is an error returned when `ValidationErr()`
+	// is called while the reader is still processing data.
+	ErrValidationIncomplete = errors.New("lfsapi/schema: validation incomplete")
```

Comment:
Do these need to be exported? Also do we need two separate types? I'm not sure an external user would care about the distinction between receiving _no data_ vs _not all of the data_.